File: lib/cms/signer_id.ts
/// <reference path="../native.ts" />
/// <reference path="../object.ts" />

namespace trusted.cms {
    /**
     * Wrap signer identifier information (keyidentifier, issuer name and serial number)
     *
     * @export
     * @class SignerId
     * @extends {BaseObject<native.CMS.SignerId>}
     */
    export class SignerId extends BaseObject<native.CMS.SignerId> {
        /**
         * Creates an instance of SignerId.
         *
         *
         * @memberOf SignerId
         */
        constructor(param?: any) {
            super();
            if (param instanceof native.CMS.SignerId) {
                this.handle = param;
            } else {
                this.handle = new native.CMS.SignerId();
            }
        }

        /**
         * Return full issuer name
         *
         * @readonly
         * @type {string}
         * @memberOf SignerId
         */
        get issuerName(): string {
            return this.handle.getIssuerName();
        }

        /**
         * Return serial number
         *
         * @readonly
         * @type {string}
         * @memberOf SignerId
         */
        get serialNumber(): string {
            return this.handle.getSerialNumber().toString();
        }

        /**
         * Return keyidentifier
         *
         * @readonly
         * @type {string}
         * @memberOf SignerId
         */
        get keyId(): string {
            return this.handle.getKeyId();
        }
    }
}

File: deps/wrapper/src/cms/signer_id.cpp
#include "../stdafx.h"

#include "wrapper/cms/signer_id.h"

void SignerId::setIssuerName(Handle<std::string> name){
	LOGGER_FN();

	this->issuerName = name;
}

Handle<std::string> SignerId::getIssuerName(){
	LOGGER_FN();

	if (!this->issuerName.isEmpty()) {
		return this->issuerName;
	}
	else {
		THROW_EXCEPTION(0, SignerId, NULL, "Issuer name is empty");
	}

	return this->issuerName;
}

void SignerId::setSerialNumber(Handle<std::string> value){
	LOGGER_FN();

	this->serialNumber = value;
}

Handle<std::string> SignerId::getSerialNumber(){
	LOGGER_FN();

	if (!this->serialNumber.isEmpty()) {
		return this->serialNumber;
	}
	else {
		THROW_EXCEPTION(0, SignerId, NULL, "Serial number is empty");
	}
}

void SignerId::setKeyId(Handle<std::string> value){
	LOGGER_FN();

	this->keyid = value;
}

Handle<std::string> SignerId::getKeyId(){
	LOGGER_FN();

	if (!this->keyid.isEmpty()) {
		return this->keyid;
	}
	else {
		THROW_EXCEPTION(0, SignerId, NULL, "KeyId is empty");
	}
}
